src/lib/support/StringCircularBuffer.h

```diff
@@ -0,0 +1,76 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstddef>
+
+#include <core/CHIPError.h>
+#include <support/Span.h>
+
+namespace chip {
+
+/**
+ * @brief A circular buffer to store strings.
+ *
+ * This circular buffer provides a queue like interface push/pop strings from
+ * the buffer. When there is no space to push a new string, the oldest string
+ * will be discarded silently to free up space.
+ *
+ * NOTE: this class never handles trailing zero.
```

Comment:
This might be clearer if this were called a ByteCircularBuffer and never talked about strings at all.